### DISCUSSION TITLE
Fangfrisch Private Database Mirror

### DISCUSSION THREAD

**Discussion by Dulux-Oz** [rseichter/fangfrisch#43] [Q&A] [answered]
Hi,

Forgive my newbie-ignorance, but is it possible to set up Fangfrisch to work with a local/priavte mirror of the various 3rd-party ClamAV Signature Databases (like ClamAV does https://docs.clamav.net/appendix/CvdPrivateMirror.html)?

Or have I got the entirely wrong end of the stick?

Thanks in advance

Dulux-Oz

**Comment by rseichter** [comment]
Fangfrisch comes with a robust set of preconfigured data sources / providers, but you are free to override or even disable these sources. If you look at [appendix A](https://rseichter.github.io/fangfrisch/#internalconf) of the documentation, for example the `[fangfrischnews]` provider section, you might get an idea of how changing the `prefix` or individual `url_foobar` settings can be used to point to alternative hosts, including private mirrors.

**Comment by Dulux-Oz** [comment]
Hi Ralph (@rseichter),

Thanks for getting back to me.

Unfortunately, because I don't think I worded my query correctly, the very good answer you provided did not answer what I wanted to know. So let me try to be more precise and have another go at it.  :-)

I currently use the ClamAV `cvdupdate` utility to download the ClamAV signature database to a private mirror on my LAN. Then the individual clients/hosts (on my LAN) contact my local mirror server via `freshclam` to grab the AV signature updates.

I gather from your answer (although I could be wrong) that `fangfrisch` can be configured to also download 3rd Party DBs from a local mirror (just like `freshclam` can).

What I was inquiring about was whether `fangfrisch` can *also* be used as the "`cvupdate` part of the process" ie can I use `fangfrisch` to download/sync the 3rd Party DBs from their sources to my local mirror (which my other hosts can then use `fangfrisch` to contact the local mirror to grab the signature updates - most likely alongside of `freshclam`).

Or do I need to use some other utility/process.

Thanks again for you help

Dulux-Oz

**Comment by rseichter** [comment, marked as answer]
On an abstract level, Fangfrisch operates by mapping remote URLs to locally stored files (as in "on some locally available file system", including network shares), with several tracking and optimisation measures added to the mix which reduce bandwidth usage and load. In other words: If X is available for download via HTTP(S) or FTP (any protocol supported by Python's Requests library), Fangfrisch can be configured to periodically download the latest version of X, store it locally, and then perform/trigger an action of your choice whenever local data was updated. This approach is flexible enough to deal with a wide variety of scenarios, but it is optimised for only a subset of possible use cases. Does this help you?
